i8_terminal/commands/metrics/metrics_historical.py

```diff
@@ -354,11 +365,27 @@ def historical(
     columns_justify: Dict[str, Any] = {}
     for metric_display_name, metric_df in df.groupby("Metric"):
         columns_justify[metric_display_name] = "left" if metric_df["display_format"].values[0] == "str" else "right"
-    formatted_df = formatted_df.pivot(
-        index=["Ticker", "Period", "PeriodDateTime"], columns="Metric", values="value"
-    ).reset_index()
-    formatted_df.sort_values(["Ticker", "PeriodDateTime"], ascending=False, inplace=True)
-    formatted_df.drop(columns=["PeriodDateTime"], inplace=True)
+    if any(set(["D", "R"]) & set(formatted_df["default_period_type"].unique())):
+        formatted_df = formatted_df.pivot(
+            index=["Ticker", "Period", "PeriodDateTime"], columns="Metric", values="value"
+        ).reset_index()
+        sorterTickerIndex = dict(zip(tickers_list, range(len(tickers_list))))
+        formatted_df["TickerRank"] = formatted_df["Ticker"].map(sorterTickerIndex)
+        formatted_df.sort_values(["TickerRank", "PeriodDateTime"], ascending=[True, False], inplace=True)
+        formatted_df.drop(columns=["TickerRank", "PeriodDateTime"], inplace=True)
+    else:
+        formatted_df = formatted_df.pivot(index=["Ticker", "Period"], columns="Metric", values="value").reset_index()
+        formatted_df["reversed_period"] = formatted_df.apply(
+            lambda row: reverse_period(row.Period),
+            axis=1,
+        )
+        sorterTickerIndex = dict(zip(tickers_list, range(len(tickers_list))))
```

Comment:
Duplicate of line 372. Move duplicate logic outside of the `if` block. What are you trying to do in this `if/else` block? Maybe it would be more readable if you move the logic to a private function.